### ISSUE TITLE
MOES Zigbee Radiator Actuator HY368

### ISSUE THREAD

**Issue by schoven** [dresden-elektronik/deconz-rest-plugin#3110]
See #3024, similar device

Was told to create a new issue, so here we go:
## Device

- Product name: MOES Zigbee Radiator Actuator. 
       - specs on package:
![image](https://user-images.githubusercontent.com/62767473/89109588-f580a400-d442-11ea-8737-30bf62fa851f.png)

- Manufacturer: _TZE200_ckud7u2l
- Model identifier: TS0601
- Device type : 
  -  Sensor
  - Thermostat

## Screenshots

![image](https://user-images.githubusercontent.com/62767473/89109301-2ad7c280-d440-11ea-90c1-aef3f93d0d8c.png)

![image](https://user-images.githubusercontent.com/62767473/89109333-84d88800-d440-11ea-9afa-b4a685d2fdce.png)

**Comment by GuillaumeP01** [comment]
hello I am also interested to have this trv in deconz because I have 10 to install.  I am available for testing if needed.

**Comment by GuillaumeP01** [comment]
zigbee2mqtt side, they have already successfully integrated [https://www.zigbee2mqtt.io/devices/TS0601_thermostat.html]
if that helps.

**Comment by Smanar** [comment]
But this device is already in deconz ?
I m seing it in the code ATM, It will be in version 80 I think.

Have you tried to compile the last code ?

**Comment by GuillaumeP01** [comment]
@Smanar 
hello thank you for your feedback.  I'm starting with homeassistant.  is your fork usable with homeassistant?  if so how to put it?  Thanks for your help.

**Comment by Smanar** [comment]
You can do nothing with HA ^^, except if you are engineer.
You haven't another machine with a real Unix OS to make test ?

**Comment by GuillaumeP01** [comment]
for testing I can mount a virtual machine with debian.  once tested will this be added to the HA plugin?

**Comment by Smanar** [comment]
I don't think you can replace yourself the file in HA, when you compile it, the result file is Hardware/sofware specific.

But If you can validate the modification, it will be added in next deconz version and too in next HA plugin version. Unfortunately for you the version 80 will be out soon, so I think the modification will be on version 81.

**Comment by GuillaumeP01** [comment]
thank you for this clarification I can wait for a new version it's not a problem.

**Comment by Smanar** [comment]
You have the procedure on github homepage https://github.com/dresden-elektronik/deconz-rest-plugin at "Install deCONZ development package (optional, Linux only)"

Just replace the step 1 by
`git clone --branch tuya2 https://github.com/Smanar/deconz-rest-plugin.git`

**Comment by GuillaumeP01** [comment]
great, thank you.
you anticipated my request
do you have to install a debian with a display server or a simple debian netinst is OK?

**Comment by Smanar** [comment]
Ha good question, never tried netinst.
As you want, the desktop is realy usefull to use deconz, but I think your device will work itself, for me it s same device just with different mac adress, so the faster/easyer way will be the better.

**Comment by GuillaumeP01** [comment]
it's good i installed everything you asked for.
what should i check now?

**Comment by Smanar** [comment]
If you have finished all the procedure, the last thing is a file replacement.

So you can remove the conbee from your production network for some time, plug it on the machine where you have the custom deconz version, and try to include the device.

For information IDK wich one devices you have on your production network, but some of them like xiaomi don't like to be out of network too long time. So if it take too long time, better to use the backup/restore feature in phoscon to make a "network clone" when you make your tries.

**Comment by GuillaumeP01** [comment]
indeed I noticed that the xiaomi sensors disconnect quickly.
i have included the TRV from phoscon from the sensor page. inclusion OK and TRV recognized in deconz as TS0601

![copie1](https://user-images.githubusercontent.com/67676892/90322623-c98f1380-df56-11ea-9f87-af2209b1aaf4.png)
![copie2](https://user-images.githubusercontent.com/67676892/90322625-ce53c780-df56-11ea-84ab-84b891758c72.png)

do you need anything else?

**Comment by Smanar** [comment]
Yep, just use it ^^, to check if you miss something.
If all is working, I will make the PR

for this device you have at least.
- low battery
- mode
- actual temperature
- setpoint

**Comment by GuillaumeP01** [comment]
Thank you for guiding me through all of these steps.

How do I check the information I receive from the TRV in deconz because it is not visible in phoscon?
Obviously there must be information:
child lock
window detection
valve detection
% battery
fashions
local temperature
current heating setpoint

And how do you send, for example, a new temperature setpoint?

**Comment by GuillaumeP01** [comment]
I did an integration in jeedom and here is the result:

Control of the setpoint: OK
on the other hand, the possible range is 5 to 35 ° C and the piloting generates an error beyond 32 ° C
Switch on / off: OK
![copie3](https://user-images.githubusercontent.com/67676892/90332372-8cf60300-dfbc-11ea-9fac-876ee8046de0.png)

I continue my tests but it seems super promising to me

**Comment by GuillaumeP01** [comment]
this is missing:

- child lock (info and action)
- window detection (info and action)
- % battery

is it possible to add them?

**Comment by Smanar** [comment]
Ha you are from jeedom ? first time I see some one testing a thermostat on it ^^
BTW, if the plugin don't support all feature yet, you have a button to display node (noeud), inside you will see all filed, event the one the device don't use yet.

You have a "lowbattery" filed , this device don't provide a battery level (or haven't see it yet).
For the 2 other I can add them, but are you sure you need them ?

**Comment by GuillaumeP01** [comment]
Do you have a screenshot to display the nodes and see all the fields?
For the 2 missing fields I think so.
For children it's pretty good and the detection of window opening I think it's useful too.

**Comment by GuillaumeP01** [comment]
by the way, is it possible to rename the devices in deconz to find them more easily when they are not displayed in phoscon?

**Comment by Smanar** [comment]
I think they called that "informations brutes" > https://community.jeedom.com/t/deconz-sondes-aqara-recuperer-des-informations-brutes/22467

The children protection was already in the Json, have added the window open state
Children protection > config/locked
Windows > state/windowopen

You need te include the device again to enable the window, no need to delete it.

**Comment by GuillaumeP01** [comment]
yes indeed I found the raw information in jeedom. I was thinking more of the raw deconz info.
in the raw jeedom info the child lock goes back up after adding a new info command and 2 action commands
On the other hand, I do not see the window detection?
![copie5](https://user-images.githubusercontent.com/67676892/90337410-98f4bb80-dfe2-11ea-9fc0-f3ecd10ed93c.png)
![copie6](https://user-images.githubusercontent.com/67676892/90337453-c80b2d00-dfe2-11ea-8b86-2fd7d8fa15a8.png)

**Comment by Smanar** [comment]
>the child lock goes back up after adding a new info command and 2 action commands
It mean it works or not ?

And no, the windows detection is not here, you have tried to re-include the device ?

**Comment by GuillaumeP01** [comment]
Sorry that was not clear but yes the child lock works with the addition of commands.

I tried to re-include the valve but it didn't change anything.

**Comment by Smanar** [comment]
Ok there is something strange, the fileld "state/windowopen" is created in same time than the "config/preset", I don't see why it s not displayed.

For battery, it s device dependent, I will enable the battery level tomorrow for your to check if this one have it.

**Comment by GuillaumeP01** [comment]
OK fine

For information there is a parameter which is not very explicit and which has 2 possible actions (on / off) is it maybe this but which is badly recorded? see screenshot
![copie8](https://user-images.githubusercontent.com/67676892/90341632-64dcc300-e001-11ea-9bcf-574dca3561a0.png)
![copie7](https://user-images.githubusercontent.com/67676892/90341638-6ad2a400-e001-11ea-804c-b4cd564c31b6.png)
I do not see any refresh or update possible, is this normal?

by the way, is it possible to rename the devices in deconz to find them more easily when they are not displayed in phoscon?

**Comment by ebaauw** [comment]
As with any Zigbee sensor, the `state` attributes are read-only.  `on` is derived from `valve` >0.

To change the name, use the API.   Do a PUT to `/sensors/21` with a body of `{"name": "Cool New Name"}`.

**Comment by ebaauw** [comment]
What version of the API plugin are you using?  `config.schedule` should be `{}` in v2.05.80 instead of `[]`.

**Comment by GuillaumeP01** [comment]
> What version of the API plugin are you using? config.schedule should be {} in v2.05.80 instead of []. 

yes this is the version
![copie9](https://user-images.githubusercontent.com/67676892/90343680-28b25e00-e013-11ea-9bf0-0c2ea228e157.png)


On the other hand, I didn't understand to change the name, do you have screenshots to guide me?

**Comment by Smanar** [comment]
Ok So I have corrected the "window open" setting. It seem on your device, you haven't return, it s just a setting that you can enable or not, so I have moved this field from state to config.

I have add to the battery field, but we have never see this value on logs  on other tuya device, so just make test for at least 24h.

To change name you can use Curl with
`curl -H 'Content-Type: application/json' -X PUT -d '{"name": "Cool New Name"}' http://PORT:IP/api/API_KEY/sensors/ID/`

**Comment by GuillaumeP01** [comment]
Perfect thank you do I need to update?

**Comment by Smanar** [comment]
Yep, pls, and try to re-include the device, BTW I m a little busy atm so I don't test on my side, so if you have error during compliation, you can stop all the procedure.

**Comment by GuillaumeP01** [comment]
The update went well and after reintegrating the valve the windowopen_set parameter appeared as well as battery.
I added the commands to activate windowopen and it seems to work but I don't see any change on the valve. more testing needs to be done.

I will leave 24 hours to see if the battery status changes

What is the schedule_on parameter?

**Comment by GuillaumeP01** [comment]
![copie10](https://user-images.githubusercontent.com/67676892/90434244-6378cc80-e0cd-11ea-9858-75cf83d344e2.png)

**Comment by Smanar** [comment]
For schedule you have an exemple here : https://github.com/dresden-elektronik/deconz-rest-plugin/issues/2393

I will ask for battery on the other issue, but if we don't find it, I think the only one solution is putting deconz in debug mode, to find the good vlaue.

**Comment by SwoopX** [comment]
What's `config/windowopen_set`? We already have `windowopen` as state...

**Comment by Smanar** [comment]
There is 2 "windows open"
- One that return value > State/windowopen (only readable)
- One that you can only set to on or off config/windowopen_set (readable and writable)

**Comment by GuillaumeP01** [comment]
After almost 24 hours I still have no feedback for the battery.

The doc indicates a low battery indicator suddenly I think there is no battery recovery.
![copie11](https://user-images.githubusercontent.com/67676892/90540329-603f1880-e181-11ea-83de-699238ada20e.png)
![copie12](https://user-images.githubusercontent.com/67676892/90544445-7059f680-e187-11ea-92a2-757269016fdf.png)

**Comment by Smanar** [comment]
Yep I think it's normal, I have an aswer just ATM

>yes, on one of my valves sometimes the bootstrapping message about lowbattery is caught, on the other never. If the battery is ok the lowbattery=false message is sent only once when you insert new batteries.
If batteries are ok, no message will be sent until the batteries go empty.
However, when the batteries go empty/dead, the lowbattery status start flapping.
So, it would be great if the default value for the lowbattery status could be "false" instead of "null" at beginning for this kind of devices. But I'm not sure is possible nor it's correct.

So you will have "Battery" =  "false", on the future version (Already online on my fork but better to wait for some days, big change ATM)

**Comment by GuillaumeP01** [comment]
do you have an example code for the schedule [] ?

**Comment by Smanar** [comment]
Reachable  = online.
For "on" , there is 2 one ?

For preset, if you want to test on your device, if it support them 
```
                    if (presetSet == "holiday") { data = QByteArray("\x00",1); }
                    else if (presetSet == "auto") { data = QByteArray("\x01",1); }
                    else if (presetSet == "manual") { data = QByteArray("\x02",1); }
                    else if (presetSet == "confort") { data = QByteArray("\x03",1); }
                    else if (presetSet == "eco") { data = QByteArray("\x04",1); }
                    else if (presetSet == "boost") { data = QByteArray("\x05",1); }
                    else if (presetSet == "complex") { data = QByteArray("\x06",1); }
```

**Comment by GuillaumeP01** [comment]
Yes that's it for "On" I have 2 actions and the state

I will do some tests for preset tests thanks for the codes

Do you have the same for schedule?

**Comment by Smanar** [comment]
Schedule have be never tested on tuya. and the code not permit that yet, so it s not on top of the TODO list ^^.

config/on is a setting : with his one you can enable or not the device (to prevent it trigger event for exemple) I think you have t on all sensor

state/on is a state : like have said Ebaawn it reflect the valve state (it is "on" when valve status > 3)

**Comment by GuillaumeP01** [comment]
on a screenshot taken from the internet we see that there are 2 additional parameters for the opening command window how do we add them?
![copie13](https://user-images.githubusercontent.com/67676892/90682829-df0d8180-e265-11ea-986a-aba6489f2d61.png)

I added the 7 presets and change of course on the trv

**Comment by GuillaumeP01** [comment]
low battery I think it works but I have no hs battery to test

**Comment by Smanar** [comment]
I have find for windows parameter, the information is visible on the device ?
I m trying to find a way to don't overload the JSON.

**Comment by GuillaumeP01** [comment]
Yes the informations for Windows parameter is visible on the device.

**Comment by Smanar** [comment]
Ok so I can made an hidden command to change this setting.
So with that you will see the values on the device instead of in the API, and no more field in the JSON.

**Comment by GuillaumeP01** [comment]
ok tell me when i can test it

**Comment by Smanar** [comment]
You have the procedure here https://github.com/dresden-elektronik/deconz-rest-plugin at "Install deCONZ development package (optional, Linux only)"

Just replace the step 1 by
`git clone --branch tuya2 https://github.com/Smanar/deconz-rest-plugin.git`

Can skip "git checkout -b mybranch HEAD"

**Comment by GuillaumeP01** [comment]
Ah ok sorry I did not understand
I update and keep you posted

**Comment by Smanar** [comment]
Ha, I think you have error displayed before this message, on diffrent color

**Comment by GuillaumeP01** [comment]
I have this:

de_web_plugin.cpp: In member function ‘void DeRestPluginPrivate::addSensorNode(const deCONZ::Node*, const deCONZ::NodeEvent*)’:
de_web_plugin.cpp:4419:84: error: expected ‘;’ before ‘{’ token
                     else (node->nodeDescriptor().manufacturerCode() == VENDOR_NONE)

**Comment by Smanar** [comment]
Ok c'est ma faute, j'ai corrigé, tu peux re tenter.

**Comment by GuillaumeP01** [comment]
tu es français ?

**Comment by GuillaumeP01** [comment]
It's good the compilation, it went well this time
I re-included the TRV and here is the new raw information:



```
    Configuration
    Informations brutes

{
    "21": {
        "config": {
            "heatsetpoint": 2350,
            "locked": false,
            "mode": "auto",
            "offset": -100,
            "on": true,
            "preset": "manual",
            "reachable": true,
            "schedule": [],
            "schedule_on": false,
            "windowopen_set": false
        },
        "ep": 1,
        "etag": "75c86eb775165cfb0a65664ea20e7405",
        "lastseen": "2020-08-23T18:31Z",
        "manufacturername": "_TZE200_ckud7u2l",
        "modelid": "TS0601",
        "name": "TS0601",
        "state": {
            "lastupdated": "none",
            "lowbattery": null,
            "on": null,
            "temperature": 2400,
            "valve": null
        },
        "type": "ZHAThermostat",
        "uniqueid": "84:2e:14:ff:fe:f1:54:bb-01-0201"
    }
}


```

**Comment by GuillaumeP01** [comment]
For the window opening I added the time and temperature settings but it did not change on the valve.
`
01-0201.config::windowopen_set::1,20,20`

**Comment by Smanar** [comment]
you need to use the bracket.

**Comment by GuillaumeP01** [comment]
can the null value be because of jeedom?
can i see some other way?

**Comment by Smanar** [comment]
I have checked the code, and I m sure you can't have "null", so not sure the procedure to change the lib have worked, you have restart deconz, after having changed the file ?

I realy don't know how work jeedom, but you can mimic the command to change color "XY"
Or use curl

`curl -H 'Content-Type: application/json' -X PUT -d '{"windowopen_set":[true,20,20]}' http://IP:PORT/api/KEY/sensors/21/config`

If something is wrong you will have an error message using curl. IP and PORt  are the same than for phoscon, for the KEY you have it in jeedom confog (can use the same)

**Comment by GuillaumeP01** [comment]
yes i restarted deconz. I even restarted my machine.
I did not have any new errors during the procedure. I can do it again if necessary. but first I'll test the curl command and let you know.
```

{
    "21": {
        "config": {
            "heatsetpoint": 2300,
            "locked": false,
            "mode": "auto",
            "offset": -100,
            "on": true,
            "preset": "manual",
            "reachable": true,
            "schedule": [],
            "schedule_on": false,
            "windowopen_set": false
        },
        "ep": 1,
        "etag": "d3f76a92a1b12ab0d045844cdcd4a48b",
        "lastseen": "2020-08-24T16:44Z",
        "manufacturername": "_TZE200_ckud7u2l",
        "modelid": "TS0601",
        "name": "TS0601",
        "state": {
            "lastupdated": "none",
            "lowbattery": null,
            "on": true,
            "temperature": 2250,
            "valve": 100
        },
        "type": "ZHAThermostat",
        "uniqueid": "84:2e:14:ff:fe:f1:54:bb-01-0201"
    }
}
```

**Comment by GuillaumeP01** [comment]
The curl command works and changes parameters on the valve well.
thank you so much

**Comment by GuillaumeP01** [comment]
I just redo the install procedure, restart the PC and it's always the same thing
All state values are null.
```

"state": {
            "lastupdated": "none",
            "lowbattery": null,
            "on": null,
            "temperature": null,
            "valve": null
```

**Comment by Smanar** [comment]
Ok, so no more idea for this problem, I will ask to other ...

**Comment by GuillaumeP01** [comment]
after the feedback, there is only low battery on null
```

"state": {
            "lastupdated": "none",
            "lowbattery": null,
            "on": false,
            "temperature": 2150,
            "valve": 0
```

**Comment by Smanar** [comment]
I have perhaps an answer from ebaawn.
It seem the value are  overwritten with the values from the database shortly after. so if you already have the value "null" in the database it can be an explanation.

You don't have deleted the device, just re-included ?

**Comment by GuillaumeP01** [comment]
yes that's it, no deletion just re-included the trv

**Comment by Smanar** [comment]
Ok so it s a possible reason, will see later, not realy dramatic ^^

**Comment by GuillaumeP01** [comment]
if you want I can do one by removing the trv and re-included it

**Comment by GuillaumeP01** [comment]
I did a test by removing the TRV but it doesn't change anything there is always null on opening window

```
 "state": {
            "lastupdated": "none",
            "lowbattery": null,
            "on": true,
            "temperature": 2450,
            "valve": 65
```

**Comment by killeriq** [comment]
Adding also details pics + link https://www.moeshouse.com/collections/trv-radiator-valve/products/tuya-zigbee3-0-new-mini-wifi-radiator-actuator-smart-programmable-thermostat-temperature-trv-controller

![IMG_20200827_223955](https://user-images.githubusercontent.com/39062891/91492897-b6653780-e8b6-11ea-8dbd-827f838fd036.jpg)

**Comment by schoven** [comment]
The work you guys put in is very much appreciated. Once finishes PM me so i can gift you guys a digital cup of Coffee :)

**Comment by GuillaumeP01** [comment]
despite several attempts, I failed to create the commands with the parameters for the window opening detection in jeedom.
After having tested for a good week, the operation chosen for me will be the auto mode and the manual preset because in jeedomle plugin thermostat really allows a lot of things.

Could you tell me Smanar when the integration of TRV in the stable branch will be available?

**Comment by Smanar** [comment]
The PR is submitted, need to wait if it will be accepted, the next deconz version will be in 10 days.
Do you use deconz in stand alone or the one integrated in the jeedom plugin ?

For the command use in jeedom, you can try to compare the values send using the log, but I remember another user have failed to use the "stop" command for a roller shutter too. something not possible to create the JSON.

**Comment by GuillaumeP01** [comment]
I use deconz from home assistant as a gateway and jeedom as a receiver deconz is not installed locally on jeedom.
I noticed that the home assistant version was updated more regularly but I may be wrong.

I tried to see the values send from the log but even if I hardened [true, 20,20] only true is sent.

Thank you very much for all your help

**Comment by Smanar** [comment]
Take a look how jeedom change the color, it need to use a json like [x,y]

**Comment by GuillaumeP01** [comment]
I looked at this track but in jeedom there is a color function directly
![copie16](https://user-images.githubusercontent.com/67676892/92302817-5e41cb80-ef6f-11ea-8ac4-0c71780f7872.png)

**Comment by Smanar** [comment]
Have you ask to the plugin dev ?
As value you can use bool (true/false) number , string just need to ask how to use array.

**Comment by GuillaumeP01** [comment]
I made the request and it is incurring of addition

**Comment by Abbadon89** [comment]
> deconz version will be in 10 days.

Where i can find this information?

**Comment by Smanar** [comment]
I don't find a link for the information, about beta/alpha/release version.

But you have at least one version every 15th of the month

**Comment by Mimiix** [comment]
@Smanar Readme file contains it.

**Comment by Smanar** [comment]
The sheduled date ?

**Comment by Abbadon89** [comment]
`This is a beta release. As recently decided in Discord, beta releases are now scheduled for every 15th of a month.`

**Comment by Mimiix** [comment]
@Smanar @Abbadon89 

@manup didn't merge my PR yet: 
https://github.com/dresden-elektronik/deconz-rest-plugin/pull/3241

**Comment by GuillaumeP01** [comment]
this is the window open command works with jeedom

see this
https://community.jeedom.com/t/vanne-thermostatique-ts0601/35691/13

**Comment by Smanar** [comment]
Lol, it s the first time I see a plugin using deconz more in date than deconz itslef ^^.

**Comment by dmatora** [comment]
Is it supposed to work on 2.07.01?

**Comment by schoven** [comment]
It does work with latest, and previous release. Still a little buggy with sensor reporting but there are People working on it & posting issue topic.

I have it heating my bathroom 45mins before my alarm wakes me up. Opening/closing works perfectly.

**Comment by dmatora** [comment]
> It does work with latest, and previous release.

Is 2.07.01 currently latest or are you talking about some sort of beta?
How do I add it? Should I scan for a light, switch or a sensor?
I'm having a hard time adding it, maybe I should just wait for new batteries to arrive

**Comment by schoven** [comment]
User instructions are in the box of the trv. In phoscon add it as a sensor(it is battery powered) . When phoscon is searching put the trv in pairing mode and it will be picked up.

**Comment by Smanar** [comment]
Tuya need "special" inclusion, It s a long story, but you can need  2 inclusions (without deleting the device), or a deconz restart if the manufacture name is good, but not the device itself.

**Comment by dmatora** [comment]
@Smanar thanks, after a few attempts I managed to enter pairing mode properly and got it added.
And you were right, I had to add it twice. After the first time it wasn't showing current temperature and was ignoring all commands.

**Comment by Oldarich** [comment]
Hello, 
short question: Does somebody know if it is possible to switch these HY368 (Tuya) to Europien Zigbee standard frequency 868MHz? I'd like to pair them with Raspberry Pi and Sonoff switching unit for controiling of heating boiler. Thank you for reply!

**Comment by Smanar** [comment]
What is the other frequency than  "Europien Zigbee standard frequency 868MHz" ?

**Comment by dmatora** [comment]
@Oldarich 868MHz is zwave frequency, not Zigbee

**Comment by Oldarich** [comment]
@dmatora and @Smanar, gents thank you for quick reply. Ok, it means that Wikipedia doesn't have accurate info, again...
[link](https://en.wikipedia.org/wiki/Zigbee) 
I hope that HY368 with 2,4GHz will cover our house completelly. I suppose that these TRVs will create P2P network.

**Comment by Smanar** [comment]
Nope, they are battery powered (not router) so they can use the mesh but don't contribute to it.

**Comment by Oldarich** [comment]
If I have one Raspberry Pi with Zigbee module and e.g. 10pcs of HY368 installed through house, does each HY368 needs direct wireless connection to Raspabery Pi or the farest HY368 will communicate via remaing HY368s? 2.4GHz with 100mW is sufficient from room to room but not from farest room to celar with boiler ;-)

**Comment by Smanar** [comment]
It can work, but not recommanded, zigbee is a meshed network, it need a mesh to work perfectly.

>the farest HY368 will communicate via remaing HY368

Nope thoses TRV are battery powered, can only be connected to router/coordinator.

**Comment by dmatora** [comment]
> I hope that HY368 with 2,4GHz will cover our house completelly. I suppose that these TRVs will create P2P network.

You need Zigbee lamps acting as retranslator

**Comment by Guus-H** [comment]
> @Smanar thanks, after a few attempts I managed to enter pairing mode properly and got it added.
> And you were right, I had to add it twice. After the first time it wasn't showing current temperature and was ignoring all commands.

@dmatora how exactlyt did you manage to get it to work? I have one paired, and until a few weeks ago it worked fine, but than it stopped listening to commands and stopped reporting the valve position etc. 
I have tried re-adding it multiple times, I have tried restarting deConz, I have tried  re-adding the deConz integration to HA but all without success. Im running deConz via HA supervisor 6.6.2 , version 2.07.01 / 8-12-2020 and Firmware 26390500. I really can't figure out what I'm missing :(

**Comment by dmatora** [comment]
Have you tried putting new batteries?

**Comment by Guus-H** [comment]
Nope, I haven't I'll try that. But as it actually does read the set temperature from the valve I don't expect too much from it, but I'll share the results

**Comment by Smanar** [comment]
I m a little lost with all tuya stuff, but there is a PR that solve missing request from TRV, there is an isue about that, but can't find it again, the PR is this one from my memory https://github.com/dresden-elektronik/deconz-rest-plugin/pull/4087

**Comment by Guus-H** [comment]
> Have you tried putting new batteries?

Doesn't work :(

**Comment by Guus-H** [comment]
> I m a little lost with all tuya stuff, but there is a PR that solve missing request from TRV, there is an isue about that, but can't find it again, the PR is this one from my memory #4087

Cool, thanks, I'll wait and see

**Comment by Paul-DS** [comment]
> > @Smanar thanks, after a few attempts I managed to enter pairing mode properly and got it added.
> > And you were right, I had to add it twice. After the first time it wasn't showing current temperature and was ignoring all commands.
> 
> @dmatora how exactly did you manage to get it to work? I have one paired, and until a few weeks ago it worked fine, but than it stopped listening to commands and stopped reporting the valve position etc.
> I have tried re-adding it multiple times, I have tried restarting deConz, I have tried re-adding the deConz integration to HA but all without success. Im running deConz via HA supervisor 6.6.2 , version 2.07.01 / 8-12-2020 and Firmware 26390500. I really can't figure out what I'm missing :(

Same problem here. The heatsetpoint command was working before, but it's not working anymore. The valve position is not reported.
I have a ConBee II with firmware version 26680700 + deconz v2.09.03.

Thanks.

**Comment by Smanar** [comment]
@Paul-DS yes my fault, the last deconz version have lot of improvement for tuya stuff but break the command "mode"/"preset" for all tuya stuff with model id = TS0601

It your missing command ?

**Comment by andye2801** [comment]
I've had one of these TRVs delivered today and I think it's paired with the ZigBee network, I can't see it on the Phoscon app but there is an unknown device on the VN map view. In Home Assistant I can see an unknown device but it has no entities or anything, am I missing something or are the past couple posts pointing to something not working in an update?

**Comment by Smanar** [comment]
@andye2801 rememebr you need to make 2 inclusion/pairing without deleting the device to correclty include it.
Can you share some screenshoot if you still have problem (from the VNC map or check the model id)

**Comment by andye2801** [comment]
![image](https://user-images.githubusercontent.com/19046762/107854164-05c20480-6e12-11eb-8f26-40005d9ef702.png)
This is the device in the VNC map. I do believe I paired it twice, do I need to reset the TRV between pairing or reboot?

**Comment by Smanar** [comment]
All is fine in deconz.
And you can't see the device in HA ?
You need to see it as plug > bad inclusion (need a second inclusion), or as ZHAthermostat > good inclusion.

But you can too have a new model, it s something starting by _TYST11_ or _TZE200_

**Comment by andye2801** [comment]
I'll give that a go then, by reset do you mean the factory reset through the menus?

**Comment by Smanar** [comment]
nope, the device (the TRV) reset

**Comment by andye2801** [comment]
Stupid question, how do I actually reset the TRV? I'm trying the AA reset option from below but this seems to be having little success.
 https://drive.google.com/file/d/116rj8RwvsLJHDkL9Y6XwAXmZ3JiX6Pnb/view

**Comment by Smanar** [comment]
IDK I haven't tuya device ^^
But yes I think it s the part that start by "after the actuactor has completed self test ...."

**Comment by andye2801** [comment]
Is there anything needed on the HA side of things or should the TRV show up as a TRV with all attributes showing?

**Comment by Smanar** [comment]
From my memory HA support ZHAthermostat, you can have issue for the "mode" but all is working.
You need to see it, at least as plug if you have bad inclusion.

**Comment by Guus-H** [comment]
I also still have issues with this TRV. It used to work in te beginning, but after an update of Phoscon it stopped working. I recently installed the new version hoping it would fix it, but unfortunately it doesn't :(
I have tried repairing (many times, no success)
It still reads the TRV, and actually updates its temperature when this is changed on the TRV, but it fails to actually set target temperatures and it doesn't report valve positions etc.
I'm guessing I might get it all to work if I can unpair it from Phoscon and start from scratch, but I can't seem to pull this off.

If you have any suggestions of something I could try that would be awesome or if you get it to work please let me know how exactly, that would be much appreciated!

For reference:
This is what it looks like in Phoscon:
![image](https://user-images.githubusercontent.com/10577609/107884662-78a4ab80-6ef6-11eb-9661-34cdc81bcf28.png)
This is what it looks like in HA:
![image](https://user-images.githubusercontent.com/10577609/107884678-a0940f00-6ef6-11eb-8246-3a6a53f30686.png)

**Comment by andye2801** [comment]
All I get in Home Assistant is: 
![image](https://user-images.githubusercontent.com/19046762/107967848-8b29ee00-6fa5-11eb-81cc-3cf6c8d3b81b.png)

Am I missing something? I have gone through the connect to zigbee network steps at least 10 times now.

**Comment by Smanar** [comment]
Yep https://github.com/dresden-elektronik/deconz-rest-plugin/issues/4417
But it only impact "mode" and "preset" are you sure the "setheatpoint" is not working ?
If yes can you show log when making the command ? deconz/help/log with "info", tuya log are easy to reconize.

**Comment by Paul-DS** [comment]
I can't find any log concerning my device when using the command `/usr/bin/deCONZ -platform minimal --http-port=80 --dbg-info=2 --dbg-zdp=1 --dbg-zcl=1 --db-aps=1`. Am I using the right command ?
I tried updating my device config using Jeedom and also directly using the Deconz API.

------

I did some testing to have an exhaustive list of non-working features:
```
{
    "9": {
        "config": {
            "heatsetpoint": 2200, 			// GET OK, SET KO
            "offset": -100,				// GET OK, SET KO
            "on": true,
            "reachable": true,
            "schedule": {...},
            "schedule_on": false			// GET KO, SET KO
        },
        "ep": 1,
        "etag": "...",
        "lastseen": "2021-02-16T07:15Z",
        "manufacturername": "Heiman",
        "modelid": "TS0601",
        "name": "Thermostat 9",
        "state": {
            "lastupdated": "none",	
            "on": true,					// GET OK
            "temperature": 2100				// GET OK
        },
        "type": "ZHAThermostat",
        "uniqueid": "..."
    }
}
```
_(GET means I can get updated values from the device, SET means I can update the value through deconz)_

------

By the way, it's probably another problem, but it seems I missing some properties:
- Locked state
- Valve state
- Boost mode


Thanks

**Comment by Smanar** [comment]
@Paul-DS Wich one deconz version are you using ?
> "manufacturername": "Heiman"

Is a bug from older version, and on the next one, the device will be not reconized at all.

If you have the GUI, you can use this method to have the correct manufcature name.
Else you realy need to delete the device first.

Tuya need 2 inclusions, the first one to memorise the good manufacture name used for the second one.
So if somewhere in the database you have writted "heiman" for this device, it will never work, because deconz will use it instead the good one.

**Comment by Paul-DS** [comment]
> @Paul-DS Wich one deconz version are you using ?

I'm using the version 2.09.03, but I did the inclusion for this device few months ago, with an older version.


> If you have the GUI, you can use this method to have the correct manufacture name #3110 (comment)
> Else you realy need to delete the device first.

I removed the device via the deconz API. I tried to pair it again, but it doesn't work.
I tried the following:
- I set phoscon in permit join.
- I put my device in pairing mode.
- I repeated the previous steps ~20 times.
- I also tried to reset the device multiple time, waiting sometime 1-2 hours before turning it on again.

Should I downgrade deconz to a previous version to make the pairing work ?

You mention a method to update manually the manufacture name, but I can't find it anywhere. This would be interesting for me as I have other "Heiman" devices that I didn't removed yet.

Don't hesitate if you need any log or additional information.

Thanks

**Comment by Smanar** [comment]
So you still have "heiman" as manufacture name ? Even with 2.09.3 ?

>Should I downgrade deconz to a previous version to make the pairing work ?

No it s the reverse, it s old version that use "heiman".

>You mention a method to update manually the manufacture name, but I can't find it anywhere. This would be interesting for me as I have other "Heiman" devices that I didn't removed yet.

Yes, it s using the GUI.

>     - Set phoscon in permit jojn (add new sensor)
>     - Read again the basic attribute in deconz (2 time if needed)

When you will read the basic attribute (cluster "basic" 0x0000, button "read") you will have the real manufacture name displayed, and the code can use it too.

**Comment by GuillaumeP01** [comment]
Hello,
Everything had been working fine so far and for some time, notably after restarting my deconz gateway, there was no way to change mode (heat, on, off). Locking and unlocking works correctly.
@smanar what do you need?
![image](https://user-images.githubusercontent.com/67676892/109478962-2af17c80-7a7a-11eb-93ea-b44c4a10c286.png)

**Comment by Smanar** [comment]
Yep, can you try a more recent version ?
The problem that disable "mode" and "preset" is corrected on recent one.

**Comment by GuillaumeP01** [comment]
Ok I update and I do a test

**Comment by GuillaumeP01** [comment]
I updated to the latest version and everything looks OK.
Thank you again for all the work you do.
![image](https://user-images.githubusercontent.com/67676892/109555334-b64c2b80-7ad5-11eb-8500-98af5473ec16.png)

**Comment by swieprecht** [comment]
Hi everyone,

So, I've purchased 7 TRV's after I thought getting my test one working fine.

Initially I had issues with them staying on heat, even when they were well over the Target Temperature, then I did a reboot on Deconz and they all went from having basic options to having loads (including the presets, auto mode, valve position etc...)

Ideal I thought... but then they all went unavailable. I tried everything but the only way I could get them to reconnect was to repair each one in Phoscon app.

All working great....

Then I needed to do another reboot on Deconz to get the thermostat working... All now gone to unavailable again.

Is this a known flaw, and is there a way to fix it so I don't have to repair every time I need to reboot Deconz? I don't often reboot Deconz, but it will be really annoying if I have to do this every single time.

**Comment by GuillaumeP01** [comment]
Yes I too am encountering the same type of problem.
Looking at the json, I noticed a difference on my 9 TRVs:

5/9
`{
    "21": {
        "config": {
            "heatsetpoint": 1950,
            "locked": true,
            "mode": "auto",
            "offset": -100,
            "on": true,
            "preset": "manual",
            "reachable": true,
            "schedule": {
                "W124": [
                    {
                        "heatsetpoint": 20,
                        "localtime": "T134:00"
                    },
                    {
                        "heatsetpoint": 15,
                        "localtime": "T08:00"
                    },
                    {
                        "heatsetpoint": 15,
                        "localtime": "T11:30"
                    },
                    {
                        "heatsetpoint": 15,
                        "localtime": "T12:30"
                    },
                    {
                        "heatsetpoint": 20,
                        "localtime": "T17:30"
                    },
                    {
                        "heatsetpoint": 15,
                        "localtime": "T22:00"
                    }
                ],
                "W3": [
                    {
                        "heatsetpoint": 20,
                        "localtime": "T06:00"
                    },
                    {
                        "heatsetpoint": 15,
                        "localtime": "T08:00"
                    },
                    {
                        "heatsetpoint": 15,
                        "localtime": "T11:30"
                    },
                    {
                        "heatsetpoint": 15,
                        "localtime": "T12:30"
                    },
                    {
                        "heatsetpoint": 20,
                        "localtime": "T17:30"
                    },
                    {
                        "heatsetpoint": 15,
                        "localtime": "T22:00"
                    }
                ]
            },
            "schedule_on": false,
            "setvalve": false,
            "windowopen_set": false
        },
        "ep": 1,
        "etag": "xxxxxx",
        "lastseen": "2021-03-03T12:38Z",
        "manufacturername": "_TZE200_ckud7u2l",
        "modelid": "TS0601",
        "name": "Vanne",
        "state": {
            "lastupdated": "none",
            "lowbattery": null,
            "on": false,
            "temperature": 2150,
            "valve": 0
        },
        "type": "ZHAThermostat",
        "uniqueid": "xxxxx-01-0201"
    }
}`

3/9
`{
    "24": {
        "config": {
            "heatsetpoint": 2100,
            "locked": true,
            "mode": "auto",
            "offset": -100,
            "on": true,
            "preset": "manual",
            "reachable": true,
            "schedule": {
                "W127": [
                    {
                        "heatsetpoint": 20,
                        "localtime": "T06:00"
                    },
                    {
                        "heatsetpoint": 15,
                        "localtime": "T08:00"
                    },
                    {
                        "heatsetpoint": 15,
                        "localtime": "T11:30"
                    },
                    {
                        "heatsetpoint": 15,
                        "localtime": "T12:30"
                    },
                    {
                        "heatsetpoint": 20,
                        "localtime": "T17:30"
                    },
                    {
                        "heatsetpoint": 15,
                        "localtime": "T22:00"
                    }
                ]
            },
            "schedule_on": false,
            "setvalve": false,
            "windowopen_set": false
        },
        "ep": 1,
        "etag": "xxxxxxxxxx",
        "lastseen": "2021-03-03T12:30Z",
        "manufacturername": "_TZE200_ckud7u2l",
        "modelid": "TS0601",
        "name": "Vanne",
        "state": {
            "lastupdated": "none",
            "lowbattery": null,
            "on": true,
            "temperature": 2000,
            "valve": 40
        },
        "type": "ZHAThermostat",
        "uniqueid": "xxxx-01-0201"
    }
}`

1/9
`{
    "36": {
        "config": {
            "heatsetpoint": 2400,
            "locked": true,
            "mode": "heat",
            "offset": -100,
            "on": true,
            "preset": "manual",
            "reachable": true,
            "schedule": [],
            "schedule_on": false,
            "setvalve": false,
            "windowopen_set": false
        },
        "ep": 1,
        "etag": "0a9a7c3378af8cb0517cdafaffb25402",
        "lastseen": "2021-03-03T12:47Z",
        "manufacturername": "_TZE200_ckud7u2l",
        "modelid": "TS0601",
        "name": "Vanne Sous Sol",
        "state": {
            "lastupdated": "none",
            "lowbattery": null,
            "on": null,
            "temperature": null,
            "valve": null
        },
        "type": "ZHAThermostat",
        "uniqueid": "84:2e:14:ff:fe:ef:e7:d7-01-0201"
    }
}`

**Comment by Smanar** [comment]
@swieprecht I have answer on your other issue.
@GuillaumeP01 I m seing same information on the 3 TRV, except the third (1/9) seem not working correctly, still no temperature.

**Comment by GuillaumeP01** [comment]
the information is not identical there is an extra part for 5 of my valves:
`"W124": [ { "heatsetpoint": 20, "localtime": "T134:00" }, { "heatsetpoint": 15, "localtime": "T08:00" }, { "heatsetpoint": 15, "localtime": "T11:30" }, { "heatsetpoint": 15, "localtime": "T12:30" }, { "heatsetpoint": 20, "localtime": "T17:30" }, { "heatsetpoint": 15, "localtime": "T22:00" } ],`

and the part is the same but named differently  W3 in the first case and W127 in the second : 
`"W3": [ { "heatsetpoint": 20, "localtime": "T06:00" }, { "heatsetpoint": 15, "localtime": "T08:00" }, { "heatsetpoint": 15, "localtime": "T11:30" }, { "heatsetpoint": 15, "localtime": "T12:30" }, { "heatsetpoint": 20, "localtime": "T17:30" }, { "heatsetpoint": 15, "localtime": "T22:00" } ] }, "schedule_on": false, "setvalve": false, "windowopen_set": false },`

is this normal?

**Comment by Smanar** [comment]
Yep, this is shedule, nothing to see with the normal working mode, you have set a schedule W124 = workday on one and W3 = Saturday and Sunday on the second one.

**Comment by GuillaumeP01** [comment]
Ok merci

**Comment by bchhabra** [comment]
Hello  Guys,

I am looking to buy a new trv which supports deconz/zigbee with my home assistant installation.
@GuillaumeP01 @swieprecht @Smanar Are you happy with these trvs?
Can you please share your feedback?

Best Regards
B

**Comment by Smanar** [comment]
I don't have tuya TRV ^^. And there is at least 30/40 models, so you can found the better like the worst.

**Comment by bchhabra** [comment]
> From my memory HA support ZHAthermostat, you can have issue for the "mode" but all is working. You need to see it, at least as plug if you have bad inclusion.

Sorry about the stupid question.
What are the recommended steps to get this TRV visible in phoscon? I just got one to try out. 
I should add as a Lights or Switches or Sensor ?

**Comment by Smanar** [comment]
>I should add as a Lights or Switches or Sensor ?

Sensor, but it will work same.
And if it will not work, forced to use the GUI to see the modelID+Manufacture name, the inclusion in the zigbee network will work even if the integration in te API not. (to see if the device is supported, there is so much clones on tuya)
